pkg/services/grafana-apiserver/endpoints/request/request.go

```diff
@@ -2,25 +2,66 @@ package request
 
 import (
 	"context"
+	"fmt"
 	"strconv"
+	"strings"
 
 	"k8s.io/apiserver/pkg/endpoints/request"
 )
 
-func OrgIDFrom(ctx context.Context) (int64, bool) {
-	ns := request.NamespaceValue(ctx)
-	return ParseOrgID(ns)
+type NamespaceInfo struct {
+	// OrgID defined in namespace (1 when using stack ids)
+	OrgID int64
+
+	// The cloud stack ID (must match the value is cfg.Settings)
+	StackID string
+
+	// For namespaces that are not org-{id} or stack-{id}
+	Other string
 }
 
-func ParseOrgID(ns string) (int64, bool) {
-	if len(ns) < 5 || ns[:4] != "org-" {
-		return 0, false
+func NamespaceInfoFrom(ctx context.Context, requireOrgID bool) (NamespaceInfo, error) {
+	info, err := ParseNamespace(request.NamespaceValue(ctx))
+	if err == nil && requireOrgID && info.OrgID < 1 {
+		return info, fmt.Errorf("expected valid orgId")
+	}
+	return info, err
+}
+
+func ParseNamespace(ns string) (NamespaceInfo, error) {
+	if ns == "default" {
+		return NamespaceInfo{
+			OrgID: 1,
+		}, nil
+	}
+
+	if strings.HasPrefix(ns, "org-") {
+		id, err := strconv.Atoi(ns[4:])
+		if id < 1 {
+			return NamespaceInfo{}, fmt.Errorf("invalid org id")
+		}
+		if id == 1 {
+			return NamespaceInfo{}, fmt.Errorf("use default rather than org-1")
```

Comment:
is there a reason we need to error here since `org-1` is a valid org? it seems like it could result in extra logic in some situations

Reply:
The complicated part is going the other direction since the org to namespace mapping will now depend on configuration.